frontend/lib/messages/model/message.model.dart

```diff
@@ -1,15 +1,79 @@
 class MessageModel {
+  final List<Conversation> conversations;
+
+  MessageModel({this.conversations = const []});
+
+  factory MessageModel.fromJson(Map<String, dynamic> json) {
+    return MessageModel(
+      conversations: (json['conversations'] as List?)
+              ?.map((conv) => Conversation.fromJson(conv))
+              .toList() ??
+          [],
+    );
+  }
+}
+
+class Conversation {
+  final String id;
+  final String userId;
+  final String userName;
+  final String userImageUrl;
   final List<Message> messages;
+  final DateTime lastMessageTime;
+  final bool isUnread;
 
-  MessageModel({this.messages = const []});
+  Conversation({
+    required this.id,
+    required this.userId,
+    required this.userName,
+    required this.userImageUrl,
+    required this.messages,
+    required this.lastMessageTime,
+    this.isUnread = false,
+  });
+
+  String get lastMessagePreview {
+    if (messages.isEmpty) return '';
```

Comment:
Just a thought: would it be better if we let flutter handle the "ellipsis" logic? As in, render what ever message there is and if it overflows, you have the `TextOverflow.ellipsis` above.